File: android-app/src/main/java/com/medziku/plantwatering/data/parts/WaterLevelSensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.WaterLevelSensor;


public class WaterLevelSensorImpl extends WaterLevelSensor implements Parcelable {

    public WaterLevelSensorImpl(String name, float waterLevel) {
        super(name, waterLevel);
    }

    private WaterLevelSensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<WaterLevelSensorImpl> CREATOR = new Creator<WaterLevelSensorImpl>() {
        @Override
        public WaterLevelSensorImpl createFromParcel(Parcel in) {
            return new WaterLevelSensorImpl(in);
        }

        @Override
        public WaterLevelSensorImpl[] newArray(int size) {
            return new WaterLevelSensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getWaterLevel());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/WaterLevelSensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.WaterLevelSensor


class WaterLevelSensorImpl : WaterLevelSensor, Parcelable {

    constructor(name: String, waterLevel: Float) : super(name, waterLevel)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(waterLevel)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<WaterLevelSensorImpl> = object : Parcelable.Creator<WaterLevelSensorImpl> {
            override fun createFromParcel(`in`: Parcel): WaterLevelSensorImpl {
                return WaterLevelSensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<WaterLevelSensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
